fdbserver/storageserver.actor.cpp

```diff
@@ -5986,7 +6003,7 @@ ACTOR Future<Void> fetchKeys(StorageServer* data, AddingShard* shard) {
 			// Then dest SS waits its version catch up with this GRV version and write the data to disk.
 			// Note that dest SS waits outside the fetchKeysParallelismLock.
 			if (lastError.code() == error_code_transaction_too_old) {
-				Version grvVersion = wait(tr.getRawReadVersion());
+				Version grvVersion = wait(getRawReadVersion(&tr));
```

Comment:
If `getRawReadVersion` fails, we might want to use `std::max(shard->fetchVersion, data->version.get());`

Reply:
I think at this stage we just do not want it fail, if it fails, this fetchKeys will be blocked forever.

Reply:
`getRawReadVersion` could fail, that is independent from fetchKeys. If `getRawReadVersion` fails, we don't want to keep retrying contacting the GRV proxy, and instead, we can use `std::max(shard->fetchVersion, data->version.get())` as the fetch version.